### PR TITLE
Remove Servers to make it work reliably again.

### PR DESCRIPTION
**Pull request by nmn**
I'm not exactly sure why this started failing, but for new versions of Flow, linter-flow stopped working for me. It will still lint once in a while, but listing-on-the-fly and other features of the linter I relied on stopped working.

On the branch, I've basically removed the servers altogether, and things are working correctly again.

I'm not completely sure why we need the server code in the first place, as flow automatically starts the servers when used automatically. We could just call `flow stop` on deactivation to make sure we don't leave flow servers running?

I'm obviously not merging this, as I'd like to hear your thoughts: @Adezandee @steelbrain

### CODE REVIEW

**Comment by steelbrain** [comment]
Hi @nmn, I was against managing servers from the start, they are a bit over handling IMHO, but @Adezandee had a point when he said we shouldn't be leaving servers running around. Now your solution @nmn of calling `flow stop` on deactivate seems very interesting

**Comment by Adezandee** [comment]
HI @nmn, @steelbrain is right, that's just to clean things up after closing Atom. But calling `flow stop` seems totally fine for that ! so :+1:

**Comment by nmn** [comment]
In that case I'll add the flow stop code to the branch and ping you guys.

**Comment by Arcanemagus** [comment]
**Bump** :stuck_out_tongue:

**Comment by nmn** [comment]
This is done for now, I'll resolve conflicts and push again I guess.

**Comment by nmn** [comment]
Everything is almost done, on my end. Just have to make sure that the linter doesn't error out when the flow server is starting up.

**Comment by nmn** [comment]
One other thing to note. This needs to be a breaking change.

As of flow 0.22 (maybe earlier) there is no longer an option to check all files. Flow now requires the `/*@flow*/` at the top of the file.

As a result I've removed the `enableAll` flag. It didn't work anymore anyway.

**Comment by nmn** [comment]
Damn. Finally got the tests to pass.
@Adezandee @steelbrain A code-review would be handy here.

File: lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use babel';
+/* @flow */
+
+
+export function check(pathToFlow: string, args: Array<string>, options: Object): Promise<string> {
+  const helpers = require('atom-linter');
```
Comment (steelbrain) [inline lib/helpers.js]: It would be nice to move this out of here so `atom-linter` require is cached

Comment (nmn) [inline lib/helpers.js]: I tried, that but it seemed to cause problems earlier. Now the tests are working, let me try again.

Comment (steelbrain) [inline lib/helpers.js]: You could also do something like this
```js
let helpers
function myCoolFunction() {
  if (!helpers) {
    helpers = require('atom-linter')
  }
  return helpers.exec(...)
}
```

**Comment by nmn** [comment]
@steelbrain fixed that thing. Still works.

File: lib/index.js [hunk 2]
```diff
@@ -78,8 +73,12 @@ export default {
         const fileText = TextEditor.buffer.cachedText;
 
         // Is flow enabled for current file ?
-        if (fileText.indexOf('@flow') === -1 && !this.enableAll) {
-          return [];
+        const firstComStart = fileText.indexOf('\/*');
```
Comment (steelbrain) [inline lib/index.js]: What was wrong with the older check?

You could maybe replace the test text with `/** @flow */`
I haven't seen any other so far

Comment (nmn) [inline lib/index.js]: The old test would detect @flow in strings etc.
This is slightly stricter.

`/* @flow */` isn't a very good test as you can put this for example:
```
/*
@flow
something else 
*/
```

Comment (nmn) [inline lib/index.js]: The flow comment has to be at the top of the file, so the new test pretty safe.

Comment (nmn) [inline lib/index.js]: I can switch it back if you think this will be a problem.

Comment (steelbrain) [inline lib/index.js]: I am fine with it either way

**Comment by steelbrain** [comment]
This code LGTM and should work, merge at will after confirming that it works as intended

**Comment by nmn** [comment]
I'm merging and publishing this as I haven't see any discussion in the last week.
